### DISCUSSION TITLE
Are there plans to add support for multivariate basis functions?

### DISCUSSION THREAD

**Discussion by mihaiconstantin** [wenjie2wang/splines2#20] [Ideas]
Thank you for this great piece of work! I've been extensively using your package and cited your work in a paper we recently submitted.

This is not an issue per se, more like I am wondering if you considered adding support for multivariate basis functions (e.g., tensor-product B-splines)? Would this be hard to incorporate into `splines2`?

**Comment by wenjie2wang** [comment]
> Thank you for this great piece of work!  I've been extensively using your package and cited your work in a paper we recently submitted.

Thanks!  So glad to hear that `splines2` helps your research.

> This is not an issue per se, more like I am wondering if you considered adding support for multivariate basis functions (e.g., tensor-product B-splines)?  Would this be hard to incorporate into `splines2`?

Yes, I thought about this.  Given the univariate basis functions, it would not be hard to construct the tensor products (and their partial derivatives).  I will give more thoughts about this feature if I find an interesting application.

**Comment by mihaiconstantin** [comment]
It is great to hear this could at one point be added to `splines2`! I will keep an eye on it in the future.
